Lib/test/test_asyncio/test_server.py

```diff
@@ -266,6 +269,49 @@ async def serve(rd, wr):
         await asyncio.sleep(0)
         self.assertTrue(task.done())
 
+    async def test_close_race(self):
+
+        srv = await asyncio.start_server(lambda *_: None, socket_helper.HOSTv4, 0)
+        srv_sock = srv.sockets[0]
+        addr = srv_sock.getsockname()
+
+        # When the server is closed before a connection is handled but after the
+        # connection is accepted, then a race-condition exists between the handler
+        # transport and the server, both which will attempt to wakeup the server to set
+        # any server waiters. We can recreate race-condition by opening a connection and
+        # waiting for the server reader callback before closing the server
+        loop = asyncio.get_running_loop()
+        srv_reader, _ = loop._selector.get_key(srv_sock.fileno()).data
+        conn_task = asyncio.create_task(asyncio.open_connection(addr[0], addr[1]))
+        for _ in range(10):
+            await asyncio.sleep(0)
+            if srv_reader in loop._ready:
+                break
```

Comment:
Sorry, forgot about Windows here.

If I set this to a fixed 3 yields without relying on `if srv_reader in loop._ready`, then the test is flaky (sometimes the error won't be observed and the test will have a false-positive)

The first yield is to let the client connection to start the connection, the second for reader to be selected, and the last for the server to accept the connection. But, I'm not sure why the accept connection tasks sometimes happens much later, and hence the false-positive.